File: code/community/1301690862798901253/opting-out-of-mip-net-sdk.go
package main

import (
    "fmt"
    "log"
    "os"

    dg "github.com/deepgram-devs/deepgram-go-sdk"
)

func main() {
    apiKey := os.Getenv("DEEPGRAM_API_KEY")
    audioURL := "YOUR_AUDIO_FILE_URL" // Replace with your audio file URL

    client, err := dg.NewClient(apiKey)
    if err != nil {
        log.Fatalf("Failed to create Deepgram client: %v", err)
    }

    options := map[string]string{
        "mip_opt_out": "true", // Custom parameter
    }

    response, err := client.TranscribeAudioUrl(audioURL, options)
    if err != nil {
        log.Fatalf("Failed to transcribe audio: %v", err)
    }

    fmt.Printf("Transcription successful: %v\n", response)
}

File: code/community/1301690862798901253/opting-out-of-mip-net-sdk.py
from deepgram import Deepgram
import os
import asyncio

async def main():
    DEEPGRAM_API_KEY = os.getenv('DEEPGRAM_API_KEY')
    dg_client = Deepgram(DEEPGRAM_API_KEY)

    audio_url = "YOUR_AUDIO_FILE_URL"  # Replace with your audio file URL

    # Setting the custom parameter mip_opt_out to be True
    options = {"mip_opt_out": True}

    response = await dg_client.transcription.prerecorded({'url': audio_url}, **options)
    print('Transcription: ', response)

# Run the main function
asyncio.run(main())